workers/unity/Packages/com.improbable.gdk.core/Utility/PlayerLoopUtils.cs

```diff
@@ -0,0 +1,238 @@
+                var groups = type.GetCustomAttributes(typeof(UpdateInGroupAttribute), true);
+                if (groups.Length == 0)
+                {
+                    simulationSystemGroup.AddSystemToUpdateList(system);
+                }
+
+                foreach (var group in groups)
+                {
+                    if (!(group is UpdateInGroupAttribute groupAttribute))
+                    {
+                        continue;
+                    }
+
+                    if (!typeof(ComponentSystemGroup).IsAssignableFrom(groupAttribute.GroupType))
+                    {
+                        Debug.LogError(
+                            $"Invalid [UpdateInGroup] attribute for {type}: {groupAttribute.GroupType} must be derived from ComponentSystemGroup.");
+                        continue;
+                    }
+
+                    var groupMgr = world.GetOrCreateSystem(groupAttribute.GroupType);
+                    if (groupMgr is ComponentSystemGroup groupSys)
+                    {
+                        groupSys.AddSystemToUpdateList(world.GetOrCreateSystem(type));
+                        if (!uniqueGroup.Contains(groupAttribute.GroupType))
+                        {
+                            uniqueGroup.Add(groupAttribute.GroupType);
+                            systems.Add(groupMgr);
+                        }
+                    }
+                }
+            }
+
+            // Sort all root groups, sorts depth first
+            foreach (var system in systems)
+            {
+                var type = system.GetType();
+                if (type == typeof(InitializationSystemGroup) ||
+                    type == typeof(SimulationSystemGroup) ||
+                    type == typeof(PresentationSystemGroup) ||
+                    type.GetCustomAttribute<UpdateInSubSystemAttribute>() != null)
+                {
+                    var groupSystem = system as ComponentSystemGroup;
+                    groupSystem.SortSystemUpdateList();
+                }
+            }
+        }
+
+        public static void AddToPlayerLoop(World world)
+        {
+            var systemGroups = world.Systems.OfType<ComponentSystemGroup>();
+            var subSystemToGroup = new Dictionary<Type, List<ComponentSystemGroup>>();
+
+            // Build lookup for PlayerLoop
+            foreach (var systemGroup in systemGroups)
+            {
+                var type = systemGroup.GetType();
+                Type subSystemType;
+
+                // Hardcode the subSystem types for Unity root-systems
+                if (type == typeof(InitializationSystemGroup))
+                {
+                    subSystemType = typeof(Initialization);
+                }
+                else if (type == typeof(SimulationSystemGroup))
+                {
+                    subSystemType = typeof(Update);
+                }
+                else if (type == typeof(PresentationSystemGroup))
```

Comment:
Could replace these with a map lookup instead?

Reply:
A map for just these 3 quick if's sounds like a bit overkill? Won't be any less code either.

Reply:
I disagree

Reply:
Just a suggestion 😉 

I'd find it slightly easier to read personally - but if you are opposed to it, then its fine.

Reply:
Keeping the `if else` here